core/data/pack/README.MD

```diff
@@ -1,63 +1,45 @@
-# Proto-Pack format Version 1.1
+# Proto-Pack format Version 2.0
 
-## File
+## Header
 
- name   | type               | description
-------- | ------------------ | ------------
- magic  | `byte[9]`          | `'p'`, `'r'`, `'o'`, `'t'`, `'o'`, `'p'`, `'a'`, `'c'`, `'k'`
- header | `Chunk<Header>`    | Header chunk
- data   | `Chunk<Section>[]` | File content
+ name   | type       | description
+------- | ---------- | ------------
+ magic  | `byte[16]` | `"ProtoPack\r\n2.0\n\0"`
 
-## Chunk\<T\>
+The header contains both types of new-lines, which is common in file
+headers to detect corruption caused by automatic new-line conversions.
+The header is followed by arbitrary number of variable-sized chunks.
+Chunks can be either object instance or type definition depending on the
+sign of the `size` field (encoded as protobuf's variable-length zigzag).
 
- name  | type             | description
------- | ---------------- | ------------
- size  | `varint`         | Size of the chunk data
- data  | `byte[size]`     | `T` data
+## Object instance (size>0)
 
-## Section
+ name     | type      | description
+--------- | --------- | ------------
+ `size`   | `sint32`  | Total size of the chunk excluding this size field.
+ `parent` | `sint32`  | If == 0: New root object instance. <br /> If < 0: Relative index of chunk which is the parent. <br /> If > 0: Reserved (potentially stream index for muxing).
+ `type`   | `sint32`  | If > 0: Type index. The object has no children. <br /> If < 0: Negated type index if it may have children. <br /> If == 0: Terminates children list of the `parent`.
+ `data`   | `byte[]`  | Proto message.
 
-A section is either a type or a object:
+Objects may form a tree. Parent object is specified as back-reference to previous
+chunk (-1 is the previous chunk). The referenced chunk must also be an object and
+it must allow addition of children. Values of `parent>=0` define a root object.
```

Comment:
> Values of `parent>=0` define a root object.

In the table above parent need to be 0 to define a root object.

Reply:
The intention is that all '>=0' define root object.
Although we only ever use 0 for now, positive values are still treated a root object by decoder.
The meaning of them is unspecified as can be used for extending in the future.

Reply:
Please be consistent in the definitions. If you state that > 0 is undefined, don't partially define it elsewhere.